File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/CropType.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 作物分类实体，对应 crop_type 表。
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CropType {

    /** 类型ID */
    private Long id;

    /** 父类型ID，一级分类为空 */
    private Long parentId;

    /** 父分类名称，列表展示字段 */
    private String parentName;

    /** 分类名称 */
    private String typeName;

    /** 描述 */
    private String description;

    /** 状态：1启用 0禁用 */
    private Integer status;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}

File: smart_farm/src/api/cropType.ts
import { get, http, post } from '@/utils/request'
import type { PageInfo } from '@/api/user'

/**
 * 作物类型接口封装，对应 smart_plant 后端 /crop-type 系列接口。
 * 页面层只调用这里的函数，避免在组件中散落后端 URL。
 */
export interface CropType {
  id?: number
  parentId?: number | null
  parentName?: string
  typeName: string
  description?: string
  status?: number
  createTime?: string
  updateTime?: string
}

export interface CropTypePayload {
  parentId?: number | null
  typeName: string
  description?: string
  status?: number
}

export interface UpdateCropTypePayload extends Partial<CropTypePayload> {
  id: number
}

export interface CropTypeListParams {
  typeName?: string
  parentId?: number
  status?: number
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

/** 分页查询作物类型列表。 */
export const listCropTypes = (params: CropTypeListParams) => {
  return get<ApiResult<PageInfo<CropType>>>(
    '/crop-type/list',
    params as unknown as Record<string, unknown>,
  )
}

/** 根据 ID 查询作物类型详情。 */
export const getCropTypeById = (id: number) => {
  return get<ApiResult<CropType>>(`/crop-type/${id}`)
}

/** 新增作物类型。 */
export const addCropType = (data: CropTypePayload) => {
  return post<ApiResult<CropType>>('/crop-type/add', data as unknown as Record<string, unknown>)
}

/** 编辑作物类型。 */
export const updateCropType = (data: UpdateCropTypePayload) => {
  return http<ApiResult<CropType>>({ url: '/crop-type', method: 'put', data })
}

/** 删除单个作物类型。 */
export const deleteCropType = (id: number) => {
  return http<ApiResult<void>>({ url: `/crop-type/${id}`, method: 'delete' })
}

/** 批量删除作物类型。 */
export const batchDeleteCropTypes = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/crop-type/batch', method: 'delete', data: ids })
}

/** 启用或禁用作物类型。 */
export const updateCropTypeStatus = (id: number, status: number) => {
  return http<ApiResult<void>>({
    url: `/crop-type/${id}/status`,
    method: 'put',
    params: { status },
  })
}
